### ISSUE TITLE
执行开启ssl提取的时候，np.isnan(ssl.detach().numpy()).sum()不为0导致后续无法训练了

### ISSUE THREAD

**Issue by wickedvalley** [RVC-Boss/GPT-SoVITS#72]
切分的音频都是5s内，但是在执行开启ssl的时候，对应的4-cnhubert下无法生成目标文件。原因是代码执行到了：    if np.isnan(ssl.detach().numpy()).sum()!= 0:return，跳出了。请问什么样的数据会导致这样的跳出，我该准备满足什么样的音频才可执行成功。

**Comment by RVC-Boss** [comment]
你的显卡是16系的吗？

**Comment by Mrchen116** [comment]
把半精度关了就可以了

**Comment by sipeter** [comment]
存在同样问题，3060T显卡。

**Comment by CloudTronUSA** [comment]
我的是4090也有同样问题 关了半精度就好了

**Comment by RVC-Boss** [comment]
@CloudTronUSA @sipeter @Mrchen116 调研一下你们非16系的卡，半精度跑2步骤的推理，跑出nan的概率高吗？

**Comment by CloudTronUSA** [comment]
@RVC-Boss NAN经常出现，80%的语言都被过滤了
However，开了全精度就没事了，但是在 wav32k 下的语言有破音的问题

把 GPT-SoVITS/GPT_SoVITS/prepare_datasets/my_utils.py 里的 load_audio(file, sr) 改为
`def load_audio_safe(file, sr):
    data, sampling_rate = librosa.load(file, sr=sr)
    return np.frombuffer(data, np.float32).flatten()`
可以解决这个问题，而且（目前来说）对训练没影响。

**Comment by sipeter** [comment]
> @CloudTronUSA @sipeter @Mrchen116 调研一下你们非16系的卡，半精度跑2步骤的推理，跑出nan的概率高吗？

3060T显卡，音频质量好，语速较快，GPT-SoVITS自带的切割音频后，全部NaN，audacity切片放入output\slicer_opt，有1、2个能通过，修改全精度无效。
但有的音频就正常，无NaN问题，不知道NaN的原因是什么。

**Comment by RVC-Boss** [comment]
该问题已解决，更新代码即可。
